src/plugProjectNishimuraU/FixKabuto.cpp

```diff
@@ -1,701 +1,136 @@
 }
 
 /*
  * --INFO--
  * Address:	803007F4
  * Size:	0001A4
  */
-void FixKabuto::Obj::changeMaterial()
-{
-	/*
-	stwu     r1, -0x20(r1)
-	mflr     r0
-	stw      r0, 0x24(r1)
-	stmw     r27, 0xc(r1)
-	mr       r27, r3
-	lwz      r3, 0x180(r3)
-	lwz      r4, 0x174(r27)
-	lwz      r12, 0(r3)
-	lwz      r29, 8(r4)
-	lwz      r12, 0xe0(r12)
-	lwz      r30, 4(r29)
-	mtctr    r12
-	bctrl
-	mr       r31, r3
-	mr       r3, r29
-	lwz      r12, 0(r29)
-	lwz      r12, 0x14(r12)
-	mtctr    r12
-	bctrl
-	lwz      r4, 0x174(r27)
-	lis      r3, j3dSys@ha
-	lbz      r0, 0(r31)
-	addi     r27, r3, j3dSys@l
-	lwz      r3, 8(r4)
-	li       r28, 0
-	lwz      r3, 4(r3)
-	lwz      r4, 0x6c(r3)
-	lwz      r3, 4(r4)
-	stb      r0, 0(r3)
-	lbz      r0, 1(r31)
-	stb      r0, 1(r3)
-	lhz      r0, 2(r31)
-	sth      r0, 2(r3)
-	lhz      r0, 4(r31)
-	sth      r0, 4(r3)
-	lbz      r0, 6(r31)
-	stb      r0, 6(r3)
-	lbz      r0, 7(r31)
-	stb      r0, 7(r3)
-	lbz      r0, 8(r31)
-	stb      r0, 8(r3)
-	lbz      r0, 9(r31)
-	stb      r0, 9(r3)
-	lhz      r0, 0xa(r31)
-	sth      r0, 0xa(r3)
-	lwz      r0, 0xc(r31)
-	stw      r0, 0xc(r3)
-	lbz      r0, 0x10(r31)
-	stb      r0, 0x10(r3)
-	lbz      r0, 0x11(r31)
-	stb      r0, 0x11(r3)
-	lbz      r0, 0x12(r31)
-	stb      r0, 0x12(r3)
-	lbz      r0, 0x13(r31)
-	stb      r0, 0x13(r3)
-	lbz      r0, 0x14(r31)
-	stb      r0, 0x14(r3)
-	lbz      r0, 0x15(r31)
-	stb      r0, 0x15(r3)
-	lbz      r0, 0x16(r31)
-	stb      r0, 0x16(r3)
-	lbz      r0, 0x17(r31)
-	stb      r0, 0x17(r3)
-	lbz      r0, 0x18(r31)
-	stb      r0, 0x18(r3)
-	lbz      r0, 0x19(r31)
-	stb      r0, 0x19(r3)
-	lha      r0, 0x1a(r31)
-	sth      r0, 0x1a(r3)
-	lwz      r0, 0x1c(r31)
-	stw      r0, 0x1c(r3)
-	lwz      r3, 4(r4)
-	lwz      r0, 0x1c(r3)
-	add      r0, r31, r0
-	subf     r0, r3, r0
-	stw      r0, 0x1c(r3)
-	lwz      r3, 4(r4)
-	lwz      r0, 0xc(r3)
-	add      r0, r31, r0
-	subf     r0, r3, r0
-	stw      r0, 0xc(r3)
-	b        lbl_80300974
-
-lbl_8030093C:
-	lwz      r4, 0xc0(r29)
-	rlwinm   r3, r28, 6, 0xa, 0x19
-	rlwinm   r0, r28, 2, 0xe, 0x1d
-	add      r4, r4, r3
-	stw      r4, 0x3c(r27)
-	lwz      r3, 0x60(r30)
-	lwz      r4, 0x2c(r4)
-	lwzx     r3, r3, r0
-	lwz      r4, 0x34(r4)
-	lwz      r12, 0(r3)
-	lwz      r12, 0x24(r12)
-	mtctr    r12
-	bctrl
-	addi     r28, r28, 1
-
-lbl_80300974:
-	lhz      r0, 0x5c(r30)
-	clrlwi   r3, r28, 0x10
-	cmplw    r3, r0
-	blt      lbl_8030093C
-	lmw      r27, 0xc(r1)
-	lwz      r0, 0x24(r1)
-	mtlr     r0
-	addi     r1, r1, 0x20
-	blr
-	*/
+void Obj::changeMaterial()
+{
+	J3DModelData* modelData;
+	J3DModel* j3dModel = m_model->m_j3dModel;
+	modelData          = j3dModel->m_modelData;
+	ResTIMG* texture   = static_cast<Mgr*>(m_mgr)->getChangeTexture();
+
+	j3dModel->calcMaterial();
+	ResTIMG* newTexture;
+	J3DTexture* j3dTexture = m_model->m_j3dModel->m_modelData->m_materialTable.m_texture;
+	newTexture             = j3dTexture->_04;
+
+	newTexture->m_textureFormat     = texture->m_textureFormat;
```

Comment:
This is an inlined operator=, if you can extract this that'll clean this code up massively

Reply:
Agreed - have made an inline copyTo method instead - could probably be an overloaded operator= instead though